src/plugins/inputs/ignore.js

```diff
@@ -0,0 +1,120 @@
+"use strict";
+
+const Chan = require("../../models/chan");
+const Msg = require("../../models/msg");
+const Helper = require("../../helper");
+
+exports.commands = [
+	"ignore",
+	"unignore",
+	"ignorelist",
+];
+
+exports.input = function(network, chan, cmd, args) {
+	const client = this;
+	let target;
+	let hostmask;
+
+	if (cmd !== "ignorelist" && (args.length === 0 || args[0].trim().length === 0)) {
+		chan.pushMessage(client, new Msg({
+			type: Msg.Type.ERROR,
+			text: `Usage: /${cmd} <nick>[!ident][@host]`,
+		}));
+
+		return;
+	}
+
+	if (cmd !== "ignorelist") {
+		// Trim to remove any spaces from the hostmask
+		target = args[0].trim();
+		hostmask = Helper.parseHostmask(target);
+	}
+
+	switch (cmd) {
+	case "ignore": {
+		// IRC nicks are case insensitive
+		if (hostmask.nick.toLowerCase() === network.nick.toLowerCase()) {
+			chan.pushMessage(client, new Msg({
+				type: Msg.Type.ERROR,
+				text: "You can't ignore yourself",
+			}));
+		} else if (!network.ignoreList.some(function(entry) {
+			return Helper.compareHostmask(entry, hostmask);
+		})) {
+			hostmask.when = Date.now();
+			network.ignoreList.push(hostmask);
+
+			client.save();
+			chan.pushMessage(client, new Msg({
+				type: Msg.Type.ERROR,
+				text: `${(hostmask.nick !== "*" ? hostmask.nick : "Hostmask")} added to ignorelist`,
+			}));
+		} else {
+			chan.pushMessage(client, new Msg({
+				type: Msg.Type.ERROR,
+				text: "The specified user/hostmask is already ignored",
+			}));
+		}
+
+		break;
+	}
+
+	case "unignore": {
+		const idx = network.ignoreList.findIndex(function(entry) {
+			return Helper.compareHostmask(entry, hostmask);
+		});
+
+		// Check if the entry exists before removing it, otherwise
+		// let the user know.
+		if (idx !== -1) {
+			network.ignoreList.splice(idx, 1);
+			client.save();
+
+			chan.pushMessage(client, new Msg({
+				type: Msg.Type.ERROR,
+				text: "Successfully removed user/hostmask from ignorelist",
```

Comment:
Can we print which hostmask was removed?

Reply:
![](https://sr.ht/vhTs.png)